source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -146,6 +152,14 @@ namespace
             std::string valueString = md->getValueString();
             if (!valueString.empty())
             {
+                // Check if the value is the default value and the metadata could be ignored.
+                auto it = defaultValuedMetadata.find(name);
+                if (it != defaultValuedMetadata.end() && it->second == valueString)
```

Comment:
This makes it so we avoid writing out uivisible=true on all inputs. When the attribute is not set we assume the input is visible.

Reply:
But what if a nodedef has a value of false for uivisible and the instance has a value of true? We still need to write out uivisible="true" in that case, no? Possible I'm missing something.

Reply:
The writeNode function has existing logic to write out any input where the uivisible differ from the corresponding nodedef input. So such inputs are always written out.

So if _"no uivisible found"_ is equal to uivisible="true", then we are good. So in the case the nodedef input say uivisible="false" the existence of this node input, and the missing uivisible, means that it is overriding the nodedef input by being visible.